### PR TITLE
Update GEP process docs with more details

### PR DESCRIPTION
**Pull request by youngnick**
/kind documentation

**What this PR does / why we need it**:
This commit updates the GEP process documentation
with some more information on what is required
to move between GEP states.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rikatz** [comment]
/lgtm
unhold as you wish

thanks!

**Comment by kflynn** [comment]
Thanks @youngnick! I think this is a big win. 🙂 

/approve

I'd like another maintainer to LGTM as well.

File: geps/overview.md [hunk 1]
```diff
@@ -111,22 +111,38 @@ Please default to GitHub discussions: they work a lot like GitHub issues which
...
+"Goals" and "Non-Goals", and documenting "Who" the GEP is for, "What" the
+GEP will do, and "Why" it is needed. For this reason, the `Provisional`
+state is also sometimes called the "Who/What/Why" stage.
 
 For Memorandum GEPs, the first version of your GEP will be the only one, as
 Memorandums have only a single stage - `Accepted`.
 
-### 3. Document Implementation Details
+The `Provisional` state is different to other states, in that iteration on it
+can occur outside of the usual Gateway API release process. To put this another
```
Comment (kflynn) [inline geps/overview.md]: OK, I guess I do have one other comment: isn't this true of `Memorandum` as well? 🤔

File: geps/gep-696/index.md [hunk 2]
```diff
@@ -31,10 +39,15 @@ Write here "What" we want to do. What is the proposal aiming to do?
...
 ## Purpose (Why and Who)
 
+(Required for Provisional status and above)
+
 Write here "Why" we want to do it. What problems are being solved? What personas are
 the target of this proposal, and why will this proposal will make their lives better?
 
 ## API
+
+( Required for Implementable status and above)
```
Comment (robscott) [inline geps/gep-696/index.md]:
```suggestion
(Required for Implementable status and above)
```

File: geps/gep-696/index.md [hunk 3]
```diff
@@ -7,10 +7,14 @@
 
 ## TLDR
 
+(Required for Provisional status and above)
```
Comment (robscott) [inline geps/gep-696/index.md]: Maybe format this as so it stays hidden as a markdown comment: 
```suggestion
[required_in]: # provisional+
```

File: geps/overview.md [hunk 4]
```diff
@@ -184,26 +225,28 @@ new strategy for achieving their graduation criteria can be established. Any
...
 ### 6. Close out the GEP issue
 
 The GEP issue should only be closed once the feature has:
-- Moved to the standard channel for distribution (if necessary)
-- Moved to a "v1" `apiVersion` for CRDs
-- been completely implemented and has wide acceptance (for process changes).
+
+* Moved to the standard channel for distribution (if necessary)
+* Moved to a "v1" `apiVersion` for CRDs
+* been completely implemented and has wide acceptance (for process changes).
```
Comment (robscott) [inline geps/overview.md]: Tiny nit: Recommend consistent use or omission of `.` to end bullet points, no strong preference on which as long as it's consistent
```suggestion
* Been completely implemented and has wide acceptance (for process changes)
```

File: geps/overview.md [hunk 5]
```diff
@@ -138,7 +154,29 @@ alternatives. Be sure to document all of these in the GEP, and why we decided
...
 
-### 4. Implement the GEP as "Experimental"
+For a GEP to enter the `Implementable` phase, there are some additional
+requirements:
+
+* One or more Gateway API maintainers must agree that the GEP is in-scope
+  for the project.
+* At least two (2) implementations must agree that they are interested in
+  implementing the feature once it reaches `Experimental`. This is to ensure
+  that there's community interest outside of the GEP owner.
```
Comment (robscott) [inline geps/overview.md]: I'd suggest that this should have a time box here - ie at least 2 implementations need to commit to experimental support of the feature within ~6 months of the feature entering experimental. Without that we're going to get stuck with a lot of noise in experimental channel.

File: geps/overview.md [hunk 6]
```diff
@@ -150,10 +188,13 @@ use the `experimental` Golang build tag to denote experimental functionality.
...
-- the graduation criteria to reach `Standard` MUST be filled out
-- the GEP must have at least one Feature Name for features described inside that
+* Any API changes must be made to the Go types.
+* The graduation criteria to reach `Standard` MUST be filled out. Note that this
+  must include sufficient conformance testing at a minimum, and should include
+  any other relevant criteria for that GEP.
+* The GEP must have at least one Feature Name for features described inside that
   will need to be tested by conformance tests.
-- a proposed probationary period (see next section) must be included in the GEP
+* A proposed probationary period (see next section) must be included in the GEP
```
Comment (robscott) [inline geps/overview.md:197]: I forgot we already had this 🤦

File: geps/overview.md (hunk 5 above)
Comment (youngnick) [reply geps/overview.md]: Agreed, done.

File: geps/gep-696/index.md (hunk 3 above)
Comment (youngnick) [reply geps/gep-696/index.md]: Formatting needs to be slightly differrent, but done.

**Comment by youngnick** [comment]
@robscott has reviewed, but is on leave now. I'm going to unhold this so that the next LGTM will merge it.

/unhold

**Comment by kflynn** [comment]
@youngnick The GEP process changes look great here. 🙂 Are the changes to `site-src/implementations/v1.2.md` and `site-src/implementations/v1.3.md` intentional?

**Comment by kflynn** [comment]
Ok, I chatted with @youngnick offline about this — the changes to `site-src/implementations/v1.2.md` and `site-src/implementations/v1.3.md` are there because checks were failing, seemingly from some other `make generate` that got missed. Nick’s plan is to drop those from version control and have them be generated on the fly at publish time, and I think that sounds like a fine plan so I’m good here. Thanks @youngnick!

/lgtm
